lib/internal/timers.js

```diff
@@ -542,6 +539,10 @@ function getTimerCallbacks(runNextTicks) {
           if (destroyHooksExist())
             emitDestroy(asyncId);
         }
+
+        // The actual logic for when a timeout happens.
+        L.remove(timer);
```

Comment:
This is needed because we want the resource info to be accessible inside the `setTimeout()` callback too, as noticed by @addaleax in https://github.com/nodejs/node/issues/40961#issuecomment-979293582.

Not sure if we want to do a similar thing for `Immediate`s because of this comment above the removal step: https://github.com/nodejs/node/blob/43e127801425d0ddb1b23b86c3788603d00d6da3/lib/internal/timers.js#L426-L427
@joyeecheung do you know of a workaround?

Reply:
Introduced another variable to keep track of the active timers to fix this.